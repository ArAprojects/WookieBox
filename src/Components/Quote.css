.box {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.crawl-container {
    display: flex;
    justify-content: center;
    height: 800px;
    perspective: 400px;
    color: #feda4a;
    font-family: 'Pathway Gothic One', sans-serif;
    font-size: 500%;
    font-weight: 600;
    letter-spacing: 6px;
    line-height: 150%;
    text-align: justify;
  }

  .crawl {
  /* Position the element so we can adjust the top property in the animation */
  position: relative;
  /* Making sure the text is fully off the screen at the start and end of the animation */
  top: -6000px;
  /* Defines the skew origin at the very center when we apply transforms on the animation */
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

@keyframes crawl {
  0% {
    /* The element starts below the screen */
    top: 0;
    /* Rotate the text 20 degrees but keep it close to the viewer */
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    /* This is a magic number, but using a big one to make sure the text is fully off the screen at the end */
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}


p {
    font-family: 'distant_galaxyregular';
    text-align: center;
    font-size: 1em;
}
