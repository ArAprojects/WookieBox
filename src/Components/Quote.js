import React from 'react';
import './Quote.css'

const Quote = ({data}) => {
  console.log("quote", data)
  if (data.opening_crawl === undefined) {
    return (
      <div className="scrollText">
        <p>Loading...</p>
      </div>
    )
      }
      else
    return (
        <section className="box">
          <div className="crawl-container">
            <div className="crawl">
                <p>{data.opening_crawl}</p>
                <br/>
                <p>{data.title} {data.release_date}</p>
            </div>
          </div>
        </section>
    )
};

export default Quote;
