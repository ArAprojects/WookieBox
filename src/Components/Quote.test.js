import React from 'react'
import { shallow } from 'enzyme'
import Quote from './Quote'

describe('Quote', () => {
    it('should match the film crawl text', () => {
        const data = {title: "Return of the Jedi", release_date: "1983-05-25", opening_crawl: "Luke Skywalker has..."}
        const wrapper = shallow(<Quote 
            data={data}
            title={data.title}
            release_date={data.release_date}
            opening_crawl={data.opening_crawl}
        />)
        expect(wrapper).toMatchSnapshot()
    })

})