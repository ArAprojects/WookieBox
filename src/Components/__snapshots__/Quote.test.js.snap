// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should match the film crawl text 1`] = `
<div
  className="scrollText"
>
  <p>
    Luke Skywalker has...
  </p>
  <br />
  <p>
    Return of the Jedi
     
    1983-05-25
  </p>
</div>
`;
